amy/extrequests/forms.py

```diff
@@ -440,6 +453,31 @@ def institution_label_from_instance(obj):
         essentially works just like `Model.__str__`."""
         return "{}".format(obj.fullname)
 
+    @feature_flag_enabled("ENFORCE_MEMBER_CODES")
+    def clean_membership_info(self, request: HttpRequest) -> dict:
+        errors = dict()
+        affiliation = self.cleaned_data.get("member_affiliation")  # yes/no/unsure
+        code = self.cleaned_data.get("member_code", "")
+        if affiliation in ["yes", "unsure"] and code:
+            # ensure that code belongs to a membership
+            try:
+                Membership.objects.get(registration_code=code)
+            except Membership.DoesNotExist:
+                errors["member_code"] = ValidationError(
+                    "This code is invalid. "
+                    "Please contact your Member Affiliate to verify your code."
+                )
```

Comment:
This is a different message than suggested here: https://github.com/carpentries/amy/issues/2380#issuecomment-1505763664
> It seems you have entered an invalid member code. Please make sure you have the current year's code and do not have any typos. If the code looks correct, please proceed to submit your form. A member of our team will follow up with you if there are any problems that require your attention.

Reply:
The current error is what I was given by the workshops team more recently in Slack along with the request for hard validation of the code.
I think this message will need updating when extra validation is added anyway, to include possible reasons for the error, so I'll include it as part of my conversations with them.